static/src/javascripts/projects/common/modules/ui/cookiesBanner.js

```diff
@@ -20,24 +20,27 @@ define([
     /**
      * Rules:
      *
-     * EU visitors only
+     * UK / INT edition readers only
      * Never seen the cookie message before
      * Show once only
      * Show only on FIRST page view
      * Persist close state
      */
     function init() {
-        var EU_COOKIE_MSG = 'GU_EU_MSG',
-            euMessageCookie = cookies.get(EU_COOKIE_MSG);
-
-        if (euMessageCookie && euMessageCookie === 'unseen') {
-            var link = 'https://www.theguardian.com/info/cookies',
-                txt = 'Welcome to the Guardian. This site uses cookies, read our policy <a href="' + link + '" class="cookie-message__link">here</a>',
-                opts = {important: true},
-                cookieLifeDays = 365,
-                msg = new Message('cookies');
-            msg.show(txt, opts);
-            cookies.add(EU_COOKIE_MSG, 'seen', cookieLifeDays);
+        if (config.switches.euCookieMsg) {
+            if (config.page.edition === 'UK' || config.page.edition === 'INT') {
+                var EU_COOKIE_MSG = 'GU_EU_MSG',
+                    euMessageCookie = cookies.get(EU_COOKIE_MSG);
+                if (!euMessageCookie || euMessageCookie != 'seen') {
+                    var link = 'https://www.theguardian.com/info/cookies',
```

Comment:
I tend to think that most of the code that would hinge on the vagaries of hoisting behaviour would be weird in other ways too, e.g. using the same name on an inner and outer variable (which would be confusing). As you allude, I tend to think these ambiguities are mitigated by other good practices like keeping a function small and focused!

Once we're in ES6 land perhaps we can just put this issue to bed by banning `var` and using `const` as much as possible. (Unrelated fact: according to the spec, consts aren't supposed to be hoisted... which is actually pretty weird)